src/controllers/mainController.ts

```diff
@@ -911,6 +917,27 @@ export default class MainController implements vscode.Disposable {
 		}
 	}
 
+	public async onExecutionPlan() {
+		const planContents = vscode.window.activeTextEditor.document.getText();
+		// remove file extension from name
+		let docName = vscode.window.activeTextEditor.document.fileName;
+		console.log("File name: ", docName);
+		let lastSlash = docName.lastIndexOf('\\');
+		if (lastSlash === -1) {
+			lastSlash = docName.lastIndexOf('/');
+		}
+		docName = docName.substring(lastSlash + 1, docName.lastIndexOf('.'));
```

Comment:
Do we need to worry about files that don't have an extension?  perhaps forced into SQLPLAN mode using the language selection command.

Reply:
for now the entry point only shows up on files with .sqlplan as an extension. I can definitely add in support for other files later.

Reply:
If I paste showplan XML into an untitled text document and then change the language mode to SQL Execution Plan, when I show the control the title just says "Execution Plan".  Is that the intended behavior?

![image](https://github.com/user-attachments/assets/a1b00ed1-2b83-4741-a038-cfb9fd0daea2)